engine/schema/src/org/apache/cloudstack/storage/datastore/db/PrimaryDataStoreDaoImpl.java

```diff
@@ -409,15 +460,13 @@ public StoragePoolVO persist(StoragePoolVO pool, Map<String, String> details) {
             sc.and(sc.entity().getScope(), Op.EQ, ScopeType.ZONE);
             return sc.list();
         } else {
-            Map<String, String> details = tagsToDetails(tags);
-
-            StringBuilder sql = new StringBuilder(ZoneWideDetailsSqlPrefix);
+            StringBuilder sql = new StringBuilder(ZoneWideTagsSqlPrefix);
```

Comment:
@nvazquez what about extracting lines 463-469 to a method? Then, we can create doc and most of all test cases. This is a pretty tricky bit of code (Especially line 468).